test/cpp/end2end/xds_end2end_test.cc

```diff
@@ -4245,9 +4245,8 @@ TEST_P(LdsRdsTest, XdsRoutingClusterUpdateClustersWithPickingDelays) {
 }
 
 TEST_P(LdsRdsTest, XdsRoutingApplyXdsTimeout) {
-  // TODO(https://github.com/grpc/grpc/issues/24549): TSAN won't work here.
-  if (BuiltUnderAsan() || BuiltUnderTsan()) return;
-
+  const int64_t kToleranceMillis = 100;
+  const int64_t kTimeoutMillis = 500;
   const int64_t kTimeoutNano = 500000000;
```

Comment:
Please define `kTimeoutNano` as a function of `kTimeoutMillis`, so that the two can't accidentally wind up set to different values in the future.